### PR TITLE
list running containers.

### PR DESCRIPTION
**Pull request by zhangh43**

### CODE REVIEW

**Comment by zhangh43** [comment]
https://www.postgresql.org/docs/8.2/static/xfunc-c.html lists how to do SRF

Datum
my_set_returning_function(PG_FUNCTION_ARGS)
{
    FuncCallContext  *funcctx;
    Datum             result;
    MemoryContext     oldcontext;
    further declarations as needed

    if (SRF_IS_FIRSTCALL())
    {
        funcctx = SRF_FIRSTCALL_INIT();
        oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
        /* One-time setup code appears here: */
        user code
        if returning composite
            build TupleDesc, and perhaps AttInMetadata
        endif returning composite
        user code
        MemoryContextSwitchTo(oldcontext);
    }

    /* Each-time setup code appears here: */
    user code
    funcctx = SRF_PERCALL_SETUP();
    user code

    /* this is just one way we might test whether we are done: */
    if (funcctx->call_cntr < funcctx->max_calls)
    {
        /* Here we want to return another item: */
        user code
        obtain result Datum
        SRF_RETURN_NEXT(funcctx, result);
    }
    else
    {
        /* Here we are done returning items and just need to clean up: */
        user code
        SRF_RETURN_DONE(funcctx);
    }
}

File: src/plc_docker_curl_api.c [hunk 1]
```diff
@@ -410,3 +419,49 @@ int plc_docker_delete_container(char *name) {
...
+	sprintf(url, method, GpIdentity.dbid);
+
+	response = plcCurlRESTAPICall(PLC_HTTP_GET, url, NULL);
+	res = response->status;
+
+	if (res == 200) {
+		res = 0;
+	} else if (res >= 0) {
+		snprintf(api_error_message, sizeof(api_error_message),
+		         "Failed to list containers (return code: %d)", res);
```
Comment (paul-guo-) [inline src/plc_docker_curl_api.c]: add dbid information?

File: management/sql/plcontainer_install.sql [hunk 2]
```diff
@@ -16,6 +16,12 @@ CREATE OR REPLACE FUNCTION plcontainer_refresh_local_config(verbose bool) RETURN
 AS '$libdir/plcontainer', 'refresh_plcontainer_config'
 LANGUAGE C VOLATILE;
 
+CREATE TYPE plcontainer_summary_results AS ('SEGMENT_ID' text, 'CONTAINER_ID' text, 'UP_TIME' text, 'OWNER' text, 'MEMORY_USAGE(KB)' text);
```
Comment (paul-guo-) [inline management/sql/plcontainer_install.sql]: The name is a bit ambitious, how about "container_summary_type"?

Comment (HaozhouWang) [reply management/sql/plcontainer_install.sql]: sure, fixed

File: src/plc_docker_curl_api.c (hunk 1 above)
Comment (HaozhouWang) [reply src/plc_docker_curl_api.c]: fixed

**Pull request merged**